Client/Frontend/Fakespot/ShoppingProduct.swift

```diff
@@ -0,0 +1,71 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+/**
+ Represents a parsed product for a URL.
+
+ - Parameter id: The product id of the product.
+ - Parameter host: The host of a product URL (without www).
+ - Parameter tld: The top-level domain of a URL.
+ - Parameter sitename: The name of a website (without TLD or subdomains).
+ - Parameter valid: If the product is valid or not.
+ */
+struct Product {
+    let id: String
+    let host: String
+    let tld: String
+    let sitename: String
+}
+
+/**
+ Class for working with the products shopping API,
+ with helpers for parsing the product from a URL
+ and querying the shopping API for information on it.
+ */
+class ShoppingProduct: FeatureFlaggable {
+    private let fakespotFeature = FxNimbus.shared.features.fakespotFeature.value()
+    private let url: URL
+
+    /*
+     Creates a product.
+     - Parameter url: URL to get the product info from.
+     */
+    init(url: URL) {
+        self.url = url
+    }
+
+    private var isFakespotFeatureEnabled: Bool {
+        guard featureFlags.isFeatureEnabled(.fakespotFeature, checking: .buildOnly) else {
+            return false
+        }
+        return true
+    }
+
+    var isShoppingCartButtonVisible: Bool {
+        return product != nil && isFakespotFeatureEnabled
+    }
+
+    /**
+     Gets a Product from a URL.
+     - Returns: Product information parsed from the URL.
+     */
+
+    var product: Product? {
```

Comment:
I don't think it makes sense to have this as a computed property and have all the checks executed every time the product is accessed. The url cannot change and so we could compute this once and store it (with a lazy var for example).

Reply:
I thought so too, but on the view it looks like the url is observed with a didSet and on debugging it seems that it is set multiple times. This means the ShoppingProduct with the current implementation cannot be instantiated once and be sure the url will not be changed. I will do some more investigation how the url changes.

Reply:
For now I made it lazy. 😇